spl/src/metadata.rs

```diff
@@ -7,6 +7,7 @@ use solana_program::pubkey::Pubkey;
 use std::ops::Deref;
 
 pub use mpl_token_metadata;
+pub use mpl_token_auth_rules;
```

Comment:
Other programs have been separated with features so it would be better if this was a feature based activation too instead of getting included with the `metadata` feature.

Reply:
this is such a great idea. I was debating earlier on how I could add pNFT methods for `transfer`, `burn`... 

Having a `programmable` feature for this is a great idea. 

Moving `Token Record` account into it would also benefit from this.